### ISSUE TITLE
Add number of months (similar to date range)

### ISSUE THREAD

**Issue by Birkaransingh94** [piwik/piwik#20956]
Issue from customer:

> I am creating these embeddings for my coworkers so they can get the best experience with Matomo's integration.
>
> The following embedding code I am using is:
>
> ```
> <div id="widgetIframe"><iframe width="100%" height="350" src="https://example.matomo.cloud/index.php?module=Widgetize&action=iframe&containerId=Goal_2&disableLink=1&widget=1&moduleToWidgetize=CoreHome&actionToWidgetize=renderWidgetContainer&idSite=1&period=day&date=yesterday" scrolling="yes" frameborder="0" marginheight="0" marginwidth="0"></iframe></div>
> ```
>
> Usually, there period is set to days. I want months or weeks, but when changing the parameter "yesterday" to "month", the embedding fails and stops showing.

-----

We replied:

> Unfortunately, at the moment there is no way to customise the number of months shown on the chart. You could, however, display a range of days using:
period=range&date=previous60 or period=range&date=2023-04-01,2023-06-30

----

Required: Data should be displayed easily by selecting the of number of months from UI.

**Comment by michalkleiner** [comment]
If I understand it correctly the enhancement here would be to allow providing the param `period` with a value of `weeks` or `months`, but we'd still need to figure out how to provide how many weeks/months back, whether it's the current week/month or previous week/month etc.

Looking at the FAQ at https://matomo.org/faq/reports/embed-a-matomo-report-in-a-html-page/, we already allow for weeks and months to be used, the customer needs to adjust the `period` param, not the `date` param.

@Birkaransingh94 if there's still some enhancement needed here, please provide a clearer requirement what exactly would be needed.

When creating widgets, there are no UI controls for the period or the dates, it's about tweaking the generated iframe URL manually, so passing the FAQ to the customer should help them.

**Comment by 9joshua** [comment]
@michalkleiner The customer needed to configure his widget chart to display a specific number of months, in this case 4 to save space in his dashboard. So the period would be month and the interval 4, similar to displaying a chart with 6 monthly data points...
![image](https://github.com/matomo-org/matomo/assets/20234985/08835ee9-55fb-47b2-adc8-a2077ec6727a)

Using `period=month&date=2023-07-01` produces a chart with 24 monthly intervals by default.

Looking at the network tab in the browser when I change the interval, I see this:
![Screenshot from 2023-07-06 19-59-47](https://github.com/matomo-org/matomo/assets/20234985/913878b6-eeb6-4af6-b332-3c2fefdbeba8)

So the only work around at the moment seems to be calling the `saveViewDataTableParameters` module prior to loading the chart.

TLDR: Adding the ability to use `date=previousX` with `period=month` would be ideal.

**Comment by rcontiuk** [comment]
In addition to the number of request, I have a customer who wishes to parameterise a widget graph adding metrics, similar to the graph below:

![image](https://github.com/matomo-org/matomo/assets/49957970/49c7c4ab-4a95-40a4-a3ff-998be2a381d9)
